### ISSUE TITLE
Can the line after a `...` contain `[..]`?

### ISSUE THREAD

**Issue by smoelius** [assert-rs/trycmd#216]
Part of my test's expected stdout looks like this (note the `...` and the `[..]`):
````
stderr=```
...
     Running tests/a.rs (examples/failure/target/debug/deps/a-[..])
error: test failed, to rerun pass `--test a`
```
````
When I run the test, the actual stdout doesn't match, and I see this in the diff:
```
  24      - ...
  25      -      Running tests/a.rs (examples/failure/target/debug/deps/a-[..])
       24 +     Finished test [unoptimized + debuginfo] target(s) in 0.00s
       25 +      Running tests/a.rs (examples/failure/target/debug/deps/a-ba92ccadfb1a363b)
```
The above looks odd to me, because I would expect the `...` to match the `Finished` line.

If I remove the `Running` line from the expected stdout, the test passes (i.e., the `...` matches both the `Finished` and `Running` lines).

This suggests to me that the line after a `...` cannot contain a `[..]`(?).

Is this correct, and if so, is it expected behavior?

**Comment by epage** [comment]
At this time, the matcher needs a literal line after `...` to correctly find when the end is.  I have run into this limitation myself and would love to loosen it.  However, the code is quite complex, handling a lot of corner cases, so I have not tried to take this on myself but if someone else makes an attempt that is reasonably understandable, I *think* I'd be open to merging (we'd have to see what trade offs exist, I haven't touched this area in a while)

**Comment by smoelius** [comment]
Thank you very much for the prompt response!

**Comment by epage** [comment]
Fixed in #279
